lib/models/ObjectQueueEntry.js

```diff
@@ -198,11 +198,7 @@ class ObjectQueueEntry extends ObjectMD {
 
     toRetryEntry(site) {
         return this.clone()
-            .setReplicationBackends([{
-                site,
-                status: 'PENDING',
-                dataStoreVersionId: '',
-            }])
+            .setReplicationSiteStatus(site, 'PENDING')
```

Comment:
i think this would introduce a significant change : this may trigger a re-replication of other sites as well....
this may be OK (since either it is already replicated/pending, and would be ignored? - or it would be FAILED and should be picked up, which may not be bad?), but i fear it may introduce some side-effect or race condition.

Resetting the whole field (in the entry only, not the metadata!) avoids this kind of effect, by targeting only the specified site.

Not sure if this is an issue, but we need to think well about this and understand what is the correct/expected solution.

Reply:
`setReplicationSiteStatus` actually only changes the replication status of the specified site
https://github.com/scality/Arsenal/blob/8f63687ef3d7dfe47de747a88f1673ae29a9b6ca/lib/models/ObjectMD.ts#L1048-L1056

In the original implementation we already set the global replication status
https://github.com/scality/backbeat/blob/2ff70385a6a59a23fdeb7186474a5c2b8feb6c72/lib/models/ObjectQueueEntry.js#L207

But it is not an issue as sites with a 'COMPLETED' state are ignored
https://github.com/scality/backbeat/blob/2ff70385a6a59a23fdeb7186474a5c2b8feb6c72/extensions/replication/tasks/MultipleBackendTask.js#L1169-L1185

Reply:
I don't think this is all. Setting a single site in the list (although doing it correctly!) will make sure the entry is skipped by all other replication processors: 
* so they will not try to replicate at all, even if there is a 'gap' in the filtering. You mentioned the filter on 'COMPLETED', but it still creates a warning which not approriate; and there are other states: typically already PENDING, or in ERROR...
* so we avoid the load of refreshing the metatadata in each data processor : the faster we drop the entry, the better

There is a significant difference with the "initial" update of the object: in that case, it is added to the kafka topic so that every processor will look at it and decide if replication is needed. In this specific case, we only want the processor for the failed destination to retry, so it would be better to stick with the minimal information. And remember this is only the kafka message (e.g. designed only the target processor) and not a real copy of the real objectMd.